### ISSUE TITLE
Unauthorized: Not authorized on ... to execute command

### ISSUE THREAD

**Issue by useo6** [Automattic/mongoose#6324]
Hi,
I'm a small application that runs every 10 minutes and uses mongoose to fetch some data from my database. The first few hours everything works fine but later I'm getting the following error:

> not authorized on production to execute command { find: "articles", filter: { _id: { $in: [ ObjectId('5a9c59f599aa5311e8119d60'), ObjectId('5a9efe1169ddd928d8109591'), ObjectId('5aa386e7f828c414a0a73025') ] } }, projection: {} }

I'm using mongoose 5.0.7 and mongodb 3.4. Are there any known issues regarding this topic?

**Comment by lineus** [comment]
@useo6 what connection options if any are you passing into `mongoose.connect()` or `mongoose.createConnection()`?

**Comment by useo6** [comment]
I'm using createConnection because I need two connections.
The connections strings are nothing special.
mongodb://<user>:<pass>@<ip>:27017/production

At the moment I replaced the login credentials of my dbOwner-User with the credentials of the admin and at the moment it works without any error. But it confuses me that it sometimes work and sometimes not.

**Comment by lineus** [comment]
yeah, that's a little weird, if it was an auth issue on the db I would expect it to happen every time, and if it was the default socketTimeout of 5 minutes, I would expect it to happen on the 2nd run of the script (assuming you're not re-running the script with cron or launchd). Have you checked the mongodb jira bug queue yet? I'll have some time later to dig around if you haven't already.

**Comment by useo6** [comment]
Alright, this error also occurs when I use my mongodb-root account. I started my small application on 08.04.2018 17:20:51 and the first error occurred today at 03:54:37. I'll check the mongodb jira bug queue...

**Comment by varunjayaraman** [comment]
@useo6 yeah thats weird. you dont have multiple dbs you might be hitting right? Are your dbs load balanced? Might you have a rogue db with some invalid credentials?

**Comment by vkarpov15** [comment]
My hunch is that you're mixing up the connections somewhere. Do you have a model that is attached to both connections? Perhaps you have a model with the same name on both connections and you're mixing up the models? Also, what's the difference between the two connections you have?
